### ISSUE TITLE
Support RaiseEventAsync on Entity.Current

### ISSUE THREAD

**Issue by olitomlinson** [Azure/azure-functions-durable-extension#1474]
There is currently no convenience method on `Entity.Current` to use the `RaiseEventAsync` API to raise an event from an Entity to an Orchestration.

The current alternative is to pass in `DurableClient` as a constructor parameter into the Entity and have it available as a property on the Entity, but it feels like this boilerplate could be removed.

![image](https://user-images.githubusercontent.com/4224880/93064308-46ed9700-f66f-11ea-962c-19f4b388dc7c.png)

FYI There is already the convenience method of `StartNewOrchestration` on `Entity.Current` so I don't see why we couldn't have the convenience method for `RaiseEventAsync` as well.

**Comment by sebastianburckhardt** [comment]
I agree that we should add this, it is somewhat important to make the programming model more complete.

Using a separate `IDurableClient` is o.k. as a temporary workaround for now but it is not "clean" from a programming model perspective; meaning it does not give the usual guarantees we would like to assume, such as effectively-once and in-order delivery of events, and transactional execution.

**Comment by olitomlinson** [comment]
@sebastianburckhardt thanks!

Given what you’ve just said about the current implementation not giving the usual guarantees, is this because of using the DurableClient within the context of an Entity, or is the same true for when using DurableClient in any other context? 

I’m just trying to understand if using RaiseEventAsync within an Entity is at a greater detriment. Many thanks!

**Comment by sebastianburckhardt** [comment]
It depends on the context:
- I would not recommend using an IDurableClient inside an orchestration. You could get into trouble with async and with replay (duplicate calls, nondeterminism issues).
- I think it is o.k. to use IDurableClient from inside an activity.

**Comment by olitomlinson** [comment]
Great thanks @sebastianburckhardt  !
